app/src/main/java/a3/com/convo/activities/LoginActivity.java

```diff
@@ -189,22 +221,39 @@ public void done(ParseException e) {
     }
 
     protected void getFriendsOnApp(AccessToken access_token) {
-        // TODO: add the friends' object IDs not facebook ids
         GraphRequest request = GraphRequest.newMyFriendsRequest(
                 access_token,
                 new GraphRequest.GraphJSONArrayCallback() {
                     @Override
                     public void onCompleted(JSONArray friends, GraphResponse response) {
                         try {
-                            ParseUser user = ParseUser.getCurrentUser();
+                            final ParseUser user = ParseUser.getCurrentUser();
                             // initialize empty likes array
-                            user.put("friends", new ArrayList<String>());
+                            user.put(FRIENDS, new ArrayList<String>());
                             for (int i = 0; i < friends.length(); i++) {
                                 JSONObject friend = friends.optJSONObject(i);
-                                String name = friend.optString("name");
-                                String id = friend.optString("id");
-                                user.add("friends", id);
-                                user.saveInBackground();
+                                String name = friend.optString(NAME);
+                                final String id = friend.optString(ID);
+                                ParseQuery<ParseUser> query = ParseUser.getQuery();
+                                query.whereEqualTo(USERNAME, id);
+                                query.findInBackground(new FindCallback<ParseUser>() {
+                                    @Override
+                                    public void done(List<ParseUser> objects, ParseException e) {
+                                        if (objects != null && objects.size() == 1) {
```

Comment:
What happens if objects size is more than 1? Do we want to skip if block?

Reply:
If there are multiple users with the same facebook id, that means we somehow created duplicate users. This technically shouldn't happen because we check the server before making a new user when a user logs in with facebook. Just in case though, I added a TODO because we need to figure out how to delete the duplicate user (because Parse doesn't let us from the application).